### PR TITLE
chore: move ssh-client & legacy-daemon from core to cli only

### LINKED ISSUE #573: Move parameters from SDK to CLI
From a conversation with @gkc:
- Move `--legacy-daemon`, `--ssh-client` from noports_core to sshnoports package
- Extends the noports_core parser in sshnoports with the additional parsing

### PR DESCRIPTION
**Pull request by xavierchanth**
closes #573 

**- What I did**

- Removed legacy-daemon and ssh-client args from the noports_core package and created an extended parser for them in the CLI
- Did a minor refactor of the main file

**- How I did it**

**- How to verify it**

**- Description for the changelog**
chore: move ssh-client & legacy-daemon from core to cli only

### CODE REVIEW

File: packages/sshnoports/bin/sshnp.dart [hunk 1]
```diff
@@ -9,51 +9,62 @@ import 'package:at_utils/at_logger.dart';
...
       );
-      String homeDirectory = getHomeDirectory()!;
-      sshnp = await sshnpFromParamsWithFileBindings(
+
+      final sshnp = await createSshnp(
         params,
         atClientGenerator: (SshnpParams params) => createAtClientCli(
           homeDirectory: homeDirectory,
           atsign: params.clientAtSign,
-          namespace: '${params.device}.sshnp',
```
Comment (xavierchanth) [inline packages/sshnoports/bin/sshnp.dart:52]: This line got removed because this value is already set by SshnpCore

**Review by gkc — CHANGES_REQUESTED** [review]
Checks are failing ...

**Comment by xavierchanth** [comment]
> Checks are failing ...

Sorry, missed a cast from String to the SupportedSshClient enum type. I explicitly typed where we extract those arguments from the ArgResults now, so we at least have some form of static analysis.

File: packages/noports_core/lib/src/sshnp/models/sshnp_params.dart [hunk 2]
```diff
@@ -106,12 +102,12 @@ class SshnpParams {
       rootDomain: params2.rootDomain ?? params1.rootDomain,
       localSshdPort: params2.localSshdPort ?? params1.localSshdPort,
       listDevices: params2.listDevices ?? params1.listDevices,
-      legacyDaemon: params2.legacyDaemon ?? params1.legacyDaemon,
+      // legacyDaemon: params2.legacyDaemon ?? params1.legacyDaemon,
```
Comment (gkc) [inline packages/noports_core/lib/src/sshnp/models/sshnp_params.dart]: should be removed

File: packages/noports_core/lib/src/sshnp/models/sshnp_params.dart [hunk 3]
```diff
@@ -106,12 +102,12 @@ class SshnpParams {
...
       remoteSshdPort: params2.remoteSshdPort ?? params1.remoteSshdPort,
       idleTimeout: params2.idleTimeout ?? params1.idleTimeout,
       addForwardsToTunnel:
           params2.addForwardsToTunnel ?? params1.addForwardsToTunnel,
-      sshClient: params2.sshClient ?? params1.sshClient,
+      // sshClient: params2.sshClient ?? params1.sshClient,
```
Comment (gkc) [inline packages/noports_core/lib/src/sshnp/models/sshnp_params.dart]: should be removed

File: packages/noports_core/lib/src/sshnp/models/sshnp_params.dart (hunk 2 above)
Comment (gkc) [reply packages/noports_core/lib/src/sshnp/models/sshnp_params.dart]: Have removed

File: packages/noports_core/lib/src/sshnp/models/sshnp_params.dart (hunk 3 above)
Comment (gkc) [reply packages/noports_core/lib/src/sshnp/models/sshnp_params.dart]: Have removed

**Review by gkc — APPROVED** [review]
LGTM thanks @XavierChanth. FYI I pushed another commit to remove a couple of lines of dead commented code

**Pull request merged**